ton_client/src/debot/tests.rs

```diff
@@ -711,21 +733,101 @@ async fn test_debot_interface_call() {
 #[tokio::test(core_threads = 2)]
 async fn test_debot_4() {
     let client = std::sync::Arc::new(TestClient::new());
-    let DebotData { debot_addr, target_addr: _, keys } = init_debot4(client.clone()).await;
+    let DebotData { debot_addr, target_addr, keys } = init_debot4(client.clone()).await;
+    let target_abi = TestClient::abi(TEST_DEBOT_TARGET, Some(2));
+
+    let target_boc = download_account(&client, &target_addr).await.expect("account must exist");
+    let account: ResultOfParse = client.request(
+        "boc.parse_account",
+        ParamsOfParse {
+            boc: target_boc
+        },
+    ).unwrap();
+    assert_eq!(account.parsed["acc_type"].as_i64().unwrap(), 0);
 
     let steps = serde_json::from_value(json!([])).unwrap();
     TestBrowser::execute(
-        client,
+        client.clone(),
         debot_addr,
         keys,
         steps,
         vec![
             format!("Target contract deployed."),
+            format!("Enter 1"),
             format!("getData"),
             format!("setData(128)"),
             format!("Sign external message:"),
             format!("Transaction succeeded"),
             format!("setData2(129)"),
         ],
     ).await;
+
+    let target_boc = download_account(&client, &target_addr).await.expect("account must exist");
+    let account: ResultOfParse = client.request(
+        "boc.parse_account",
+        ParamsOfParse {
+            boc: target_boc
+        },
+    ).unwrap();
+    assert_eq!(account.parsed["acc_type"].as_i64().unwrap(), 1);
+
+    assert_get_method(
+        &client,
+        &target_addr,
+        &target_abi,
+        "getData",
+        json!({"key": 1}),
+        json!({"num": format!("0x{:064x}", 129) })
+    ).await;
+
+}
+
+async fn download_account(client: &Arc<TestClient>, addr: &str) -> Option<String> {

```

Comment:
You can use `TestClient::fetch_account`

Reply:
Ok.

Reply:
I cannot use it, because i need an optional account on return to assert that before deploy account doesn't exist (download_account returns None) and after deploy account has acc_type == 1 (Active).

Reply:
You can modify existing functions. They are not sealed :-)